pkg/recutil/reconcile.go

```diff
@@ -58,7 +59,11 @@ func ResolveAndReconcile(ctx context.Context, logger kitlog.Logger, mgr manager.
 		logger.Log("event", EventRequestStart, "msg", "Reconcile request start")
 
 		// Prepare a new object for this request
-		obj := objType.DeepCopyObject()
+		rawObj := objType.DeepCopyObject()
+		obj, ok := rawObj.(ObjWithMeta)
+		if !ok {
+			return res, errors.New("reconciled object does not have metadata")
+		}
```

Comment:
Ah interesting. Under what condition can this happen? I thought the API would reject anything without metadata?

Reply:
It should never happen (although I guess you could trigger it in tests). I just wanted to ensure we'd end the reconciliation loop with a graceful error, rather than a segfault, if this assertion ever turned out to be not true.